pubnub/models/consumer/entities/page.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABCMeta
+
+
+class PNPage:
+    __metaclass__ = ABCMeta
+
+    def __init__(self, hash):
+        self._hash = str(hash)
+
+    @property
+    def hash(self):
+        return self._hash
+
+
+class Next(PNPage):
+    def __init__(self, hash):
+        PNPage.__init__(self, hash)
+
+
+class Previous(PNPage):
+    def __init__(self, hash):
+        PNPage.__init__(self, hash)
+
+
+class PNPageable(object):
+    __metaclass__ = ABCMeta
+
+    def __init__(self, result):
+        self.total_count = result.get('totalCount', None)
+        if result.get("next", None):
```

Comment:
You can move this logic to `PNPage` using class method.
If you will decide to go with that direction your code probably will look like something like this:

```

self.next = Next.builder(result.get("next", None)
```

In the `PNPage` You will need to add:

```
@classmethod
def builder(cls, value):
       if value is None:
           return None
       return cls(value)
```